pkg/api/v1beta1/types.go

```diff
@@ -1483,6 +1487,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty" description:"host networking requested for this pod"`
+	// PullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecrets []ObjectReference `json:"pullSecrets,omitempty" description:"list of references to secrets in the same namespace available for pulling the container images" patchStrategy:"merge" patchMergeKey:"name"`
```

Comment:
> The other fields related to images are prefixed with "image", so I suggest "imagePullSecrets", like "imagePullPolicy".
> 
> Do we expect this field to be set explicitly by users, or mostly automatically?

Renamed.  I see this as a field set by a user.  I think there will be another congruent field on ServiceAccounts.  When the kubelet starts working on pod, it will chase the referenced ServiceAccount, get its list of `ImagePullSecrets`, union that list with the pod's list of `ImagePullSecrets`, and pass that list to the image puller.

Having this field settable by a user makes it possible and relatively easy to make specific pull secrets work without having to use service accounts.